verilog/tools/kythe/indexing_facts_tree_auto_pop.h

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_VERILOG_TOOLS_KYTHE_INDEXING_FACTS_TREE_AUTO_POP_H_
+#define VERIBLE_VERILOG_TOOLS_KYTHE_INDEXING_FACTS_TREE_AUTO_POP_H_
+
+#include <vector>
+
+#include "verilog/tools/kythe/indexing_facts_tree.h"
+
+namespace verilog {
+namespace kythe {
+
+namespace {
+
+template <class V, class T>
+class AutoPopBack {
```

Comment:
Normally for a common/ library like this we'd require at least

* class documentation
* unit tests

but if you don't want to be slowed down by this feel free to leave this as a `// TODO(fangism): move/refactor to common/util` for me